Correct and expand doc comments

amartini51 opened the pull request:
Add information to floating point types, per Tim Kientzle's comments.

Don't separate attribute from the rest of a declaration, which prevents proper parsing of doc comments.

Put code example in code voice.

Fixes <rdar://problem/58716408&65609221&51409021>.

amartini51 committed:
Document floating point descriptions.

Fixes <rdar://problem/51409021>.

amartini51 committed:
Put code snippet in code voice.

Fixes <rdar://problem/65609221>.

amartini51 committed:
Keep doc comment before declaration.

Because the attribute is part of the declaration, putting a doc comment
between the attribute and "public" ends up with the comment in the
middle of the declaration.  This results in SourceKit skipping over the
comment, and the docs not being shown.

Fixes <rdar://problem/58716408>.

amartini51 commented:
@swift-ci Please smoke test

tbkka commented on stdlib/public/core/FloatingPointTypes.swift.gyb:
```diff
@@ -106,6 +112,9 @@ extension ${Self}: CustomStringConvertible {
 extension ${Self}: CustomDebugStringConvertible {
   /// A textual representation of the value, suitable for debugging.
+  ///
+  /// This property has the same value as the `description` propyrty, except
```
Spelling: "property"

amartini51 replied:
Thanks, fixed in 8058088cb82.

tbkka reviewed (approved):
Thanks for clarifying the docs here.  I just saw one minor spelling error.

amartini51 committed:
Fix spelling and stray space.

amartini51 committed:
Avoid 'a' vs 'an' alternation before type name.

chuckdude commented on stdlib/public/core/FloatingPointTypes.swift.gyb:
```diff
@@ -90,6 +90,12 @@ public struct ${Self} {
   /// A textual representation of the value.
+  ///
+  /// For all finite values, the value of this property is a string that can be
+  /// converted back to an instance of ${Self} without rounding errors.  That
```
Pls add code font to ${Self}

amartini51 replied:
Fixed in f1f7422a827.

chuckdude reviewed (approved):
Approved with the minor change of styling ${Self} in code font. Otherwise, this looks good.

Thanks, Alex. :)

amartini51 committed:
Add missing code voice.

amartini51 commented:
@swift-ci Please smoke test

xwu commented on stdlib/public/core/FloatingPointTypes.swift.gyb:
```diff
@@ -90,6 +90,12 @@ public struct ${Self} {
 extension ${Self}: CustomStringConvertible {
   /// A textual representation of the value.
+  ///
+  /// For all finite values, the value of this property is a string that can be
```
This is a guarantee for all types that conform to `LosslessStringConvertible`—I think it would be nice to harmonize the text with what’s there, and perhaps even reference the protocol explicitly.

tbkka replied:
The documentation for `LosslessStringConvertible` is far from ideal.  It currently states "... it should be possible to re-create an instance from its string representation."  That sentence has two problems:  First, it's not binding (it's "should", not "must").  More importantly, it's not clear what it means to "re-create an instance."  (Does that mean that the new instance is `==` to the old one?  That would require that `LosslessStringConvertible` conform to `Equatable`.)

Deferring to that here is problematic because of NaN values.  NaN values can't be compared for equality (so it's not easy to define what it means to "re-create an instance"), and the current `description` property doesn't distinguish different NaN values.

xwu commented on stdlib/public/core/FloatingPointTypes.swift.gyb:
```diff
@@ -106,6 +112,9 @@ extension ${Self}: CustomStringConvertible {
   /// A textual representation of the value, suitable for debugging.
+  ///
+  /// This property has the same value as the `description` property, except
+  /// that NaN values are printed in an extended format.
```
I don’t know if we want to guarantee this format for all time (@steventyrone?). If so, I would describe what the “extended” format is (something like: “For debugging purposes, the representation includes any NaN payload and distinguishes between quiet and signaling NaN; for all other values, it is identical to the value’s `description`.”).

tbkka replied:
That makes sense, though I'd prefer to mention the non-NaN case first (since that's probably more common):

> This property is identical to the `description` property for any value that is not a NaN.  For improved debugging, the `debugDescription` of a NaN value also includes any NaN payload, quiet/signaling bits, and sign.

shahmishal commented:
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

amartini51 committed:
Use the correct argument label in prose.

See also these commits, which fixed the same issue in the code listing:

    589dd73d3934e944697d1558334a5dac4d7a40b0
    bbe18119506ae36e5f0079d30e2e630567b2c88b
